Lib/statistics.py

```diff
@@ -817,6 +844,110 @@ def pstdev(data, mu=None):
+    n = len(x)
+    if len(y) != n:
+        raise StatisticsError('covariance requires that both variables have same number of data points')
+    if n < 2:
+        raise StatisticsError('covariance requires at least two data points')
+    xbar = mean(x)
+    ybar = mean(y)
+    total = fsum((xi - xbar) * (yi - ybar) for xi, yi in zip(x, y))
+    return total / (n - 1)
+
+
+def correlation(x, y, /):
+    """Pearson's correlation coefficient
+
+    Return the Pearson's correlation coefficient for two variables. Pearson's
+    correlation coefficient *r* takes values between -1 and +1. It measures the
+    strength and direction of the linear relationship, where +1 means very
+    strong, positive linear relationship, -1 very strong, negative linear
+    relationship, and 0 no linear relationship.
+
+    >>> x = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    >>> y = [9, 8, 7, 6, 5, 4, 3, 2, 1]
+    >>> correlation(x, x)
+    1.0
+    >>> correlation(x, y)
+    -1.0
+
+    """
+    n = len(x)
+    if len(y) != n:
+        raise StatisticsError('correlation requires that both variables have same number of data points')
+    if n < 2:
+        raise StatisticsError('correlation requires at least two data points')
+    cov = covariance(x, y)
+    stdx = stdev(x)
+    stdy = stdev(y)
+    try:
+        return cov / (stdx * stdy)
+    except ZeroDivisionError:
+        raise StatisticsError('standard deviation of at least one of the variables is zero')
+
+
+def linear_regression(regressor, dependent_variable):
+    """Calculate intercept and slope for simple linear regression
+
+    Return the ``(intercept, slope)`` tuple of the simple linear regression
+    parameters. Simple linear regression describes relationship between
+    *regressor* and *dependent variable* in terms of linear function::
+
+        dependent_variable = intercept + slope * regressor + noise
+
+   where ``intercept`` and ``slope`` are the regression parameters that are
+   estimated, and noise term is an unobserved random variable, for the
+   variability of the data that was not explained byt the linear regression
+   (it is equal to the difference between prediction and the actual values
+   of dependent variable).
+
+    >>> regressor = [1, 2, 3, 4, 5]
+    >>> noise = NormalDist().samples(5, seed=42)
+    >>> dependent_variable = [2 + 3 * regressor[i] + noise[i] for i in range(5)]
+    >>> linear_regression(regressor, dependent_variable)  #doctest: +ELLIPSIS
+    (1.75684970486..., 3.09078914170...)
+
+    """
+    n = len(regressor)
+    if len(dependent_variable) != n:
+        raise StatisticsError('linear regression requires that both variables have same number of data points')
+    if n < 2:
+        raise StatisticsError('linear regression requires at least two data points')
+    try:
+        cor = correlation(regressor, dependent_variable)
+        stdx = stdev(regressor)
+        stdy = stdev(dependent_variable)
+        slope = cor * (stdy / stdx)
+    except ZeroDivisionError:
```

Comment:
This exception handling is unnecessary, since the call to `correlation()` will make the same check.

(If changing this as suggested above to use `covariance()` but not `correlation()`, only the final division operation would this exception handling.)